torchdrivesim/traffic_lights.py

```diff
@@ -0,0 +1,120 @@
+import json
+import random
+from typing import List
+from functools import reduce, lru_cache
+from enum import auto, Enum
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+
+class ControlStateType(Enum):
+    none = auto()
+    green = auto()
+    yellow = auto()
+    red = auto()
+
+
+class TrafficLightStateMachine:
+    def __init__(self, json_file_path: str, tick_to_second_ratio: float = 0.1):
```

Comment:
It would be nice to be able to construct this object without any json files, and to have the state machine structure reflected in code. Maybe introduce a `TrafficLightGroupState` class, have the `TrafficLightStateMachine` constructor take a list of those, and provide a separate function for loading a `TrafficLightStateMachine` from json?